lkowner.h: improve lkowner_unparse() efficiency

mykaul opened the pull request:
Given that usually lkowner data is 8 bytes long (from pointer or uint64),
    we can improve lkowner_unparse() and potentially skip some if's and sprintf().
    lkowner_unparse() is called from lkowner_utoa() which is used too often in
    logging calls. Even if those logs eventually are not emitted, the function is
    still called (to construct the full log string), so there's benefit in improving it.
    
    Signed-off-by: Yaniv Kaul <ykaul@redhat.com>

Commit:
common-utils: remove unused lkowner_utoa_r() function

Remove the unused lkowner_utoa_r() function from the code base.

Signed-off-by: Yaniv Kaul <ykaul@redhat.com>

Commit:
lkowner.h: improve lkowner_unparse() efficiency

Given that usually lkowner data is 8 bytes long (from pointer or uint64),
we can improve lkowner_unparse() and potentially skip some if's and sprintf().
lkowner_unparse() is called from lkowner_utoa() which is used too often in
logging calls. Even if those logs eventually are not emitted, the function is
still called (to construct the full log string), so there's benefit in improving it.

Signed-off-by: Yaniv Kaul <ykaul@redhat.com>

gluster-ant commented:
CLANG-FORMAT FAILURE:
Before merging the patch, this diff needs to be considered for passing clang-format

```diff --git a/libglusterfs/src/glusterfs/common-utils.h b/libglusterfs/src/glusterfs/common-utils.h
index a87403837..490d50591 100644
--- a/libglusterfs/src/glusterfs/common-utils.h
+++ b/libglusterfs/src/glusterfs/common-utils.h
@@ -1102,7 +1102,7 @@ __gf_thread_set_name(pthread_t thread, const char *name)
 #elif defined(HAVE_PTHREAD_SET_NAME_NP)
     pthread_set_name_np(thread, name);
     return 0;
-#else /* none found */
+#else  /* none found */
     return -ENOSYS;
 #endif /* set name */
 }
```

mykaul commented:
/run regression

amarts commented on libglusterfs/src/glusterfs/lkowner.h:
```diff
@@ -68,20 +91,19 @@ is_same_lkowner(gf_lkowner_t *l1, gf_lkowner_t *l2)
         if (lkowner->data[i] != 0) {
-            is_null = 0;
+            return 0;
             break;
```
nit: dead code.

mykaul replied:
Fixed

amarts reviewed (commented):
LGTM

xhernandez commented on libglusterfs/src/glusterfs/lkowner.h:
```diff
@@ -19,8 +19,29 @@ lkowner_unparse(gf_lkowner_t *lkowner, char *buf, int buf_len)
+       (a pointer) which will require 16 bytes (+ a NULL termination char)
+       long buffer to convert into.
+    */
+    if (lkowner->len >= 8 && buf_len > 16) {
```
why not do this for the entire buffer, and not only the first bytes ?

mykaul replied:
Because then I have the handle this part:
```
        if (i && !(i % 8)) {
            buf[j] = '-';
            j++;
        }
```

xhernandez commented on libglusterfs/src/glusterfs/lkowner.h:
```diff
@@ -19,8 +19,29 @@ lkowner_unparse(gf_lkowner_t *lkowner, char *buf, int buf_len)
+    */
+    if (lkowner->len >= 8 && buf_len > 16) {
+        for (i = 0; i < 8; i++) {
+            nibble = lkowner->data[i] & 0x0F;
+            if (nibble < 10)
+                buf[j] = '0' + nibble;
+            else
+                buf[j] = 'a' + nibble - 10;
+            j++;
+            nibble = (lkowner->data[i] & 0xF0) >> 4;
+            if (nibble < 10)
+                buf[j] = '0' + nibble;
+            else
+                buf[j] = 'a' + nibble - 10;
+            j++;
```
The nibbles are written in reverse order. And I think a lookup table would be faster:
```suggestion
            buf[j++] = hex_digits[lkowner->data[i] >> 4];
            buf[j++] = hex_digits[lkowner->data[i] & 15];
```

mykaul replied:
Fixed the nibble order. Unsure if I should have a table lookup (memory access) instead of an 'if' + simple assignment. Could be, but that goes beyond my nano optimizations already,

Commit:
lkowner.h: address review comments
- remove dead code ('break' statement)
- order of writting the nibbles was reversed (duh!)

Signed-off-by: Yaniv Kaul <ykaul@redhat.com>

gluster-ant commented:
CLANG-FORMAT FAILURE:
Before merging the patch, this diff needs to be considered for passing clang-format

```diff --git a/libglusterfs/src/common-utils.c b/libglusterfs/src/common-utils.c
index 1a61f4bf0..2cae611c1 100644
--- a/libglusterfs/src/common-utils.c
+++ b/libglusterfs/src/common-utils.c
@@ -5461,7 +5461,7 @@ gf_pipe(int fd[2], int flags)
     int ret = 0;
 #if defined(HAVE_PIPE2)
     ret = pipe2(fd, flags);
-#else /* not HAVE_PIPE2 */
+#else  /* not HAVE_PIPE2 */
     ret = pipe(fd);
     if (ret < 0)
         return ret;
```

mykaul commented:
/run regression
